lightningd/peer_control.c

```diff
@@ -78,44 +75,57 @@ void peer_set_condition(struct peer *peer, enum peer_state state)
 	peer->state = state;
 }
 
-static struct peer *new_peer(struct lightningd *ld,
-			     struct io_conn *conn,
-			     struct command *cmd)
+void add_peer(struct lightningd *ld, u64 unique_id,
```

Comment:
We do actually free it:

	add_peer(handshaked->ld, c->unique_id, fds[0], id, &cs);
	/* Now shut handshaked down (frees c as well) */
	return false;